// react-text-filter/lib/Component.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TextFilter = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactDebounceInput = require('react-debounce-input');

var _reactDebounceInput2 = _interopRequireDefault(_reactDebounceInput);

var _ReactComponentWithPureRenderMixin = require('react/lib/ReactComponentWithPureRenderMixin');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var TextFilter = exports.TextFilter = _react2.default.createClass({
  displayName: 'TextFilter',

  propTypes: {
    onFilter: _react2.default.PropTypes.func.isRequired,
    filter: _react2.default.PropTypes.string,
    minLength: _react2.default.PropTypes.number,
    debounceTimeout: _react2.default.PropTypes.number
  },

  getDefaultProps: function getDefaultProps() {
    return {
      minLength: 2,
      debounceTimeout: 300
    };
  },


  shouldComponentUpdate: _ReactComponentWithPureRenderMixin.shouldComponentUpdate,

  render: function render() {
    var _props = this.props;
    var onFilter = _props.onFilter;
    var filter = _props.filter;

    var props = _objectWithoutProperties(_props, ['onFilter', 'filter']);

    return _react2.default.createElement(_reactDebounceInput2.default, _extends({ placeholder: 'Filter'
      // Pass props through
    }, props, {
      onChange: onFilter,
      value: filter }));
  }
});

exports.default = TextFilter;
